nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/provider/ElasticSearchNxqlPageProvider.java

```diff
@@ -212,4 +220,47 @@ protected void incorporateAggregates(Map<String, Serializable> eventProps) {
             eventProps.put("aggregatesMatches", aggregateMatches);
         }
     }
+
+    @Override
+    public boolean isLastPageAvailable() {
+        if ((getResultsCount() + getPageSize()) < getMaxResultWindow()) {
+            return super.isNextPageAvailable();
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isNextPageAvailable() {
+        if ((getCurrentPageOffset() + 2 * getPageSize()) < getMaxResultWindow()) {
+            return super.isNextPageAvailable();
+        }
+        return false;
+    }
+
+    @Override
+    public long getPageLimit() {
+        return getMaxResultWindow() / getPageSize();
+    }
+
+    public long getMaxResultWindow() {
+        if (maxResultWindow == null) {
+            ConfigurationService cs = Framework.getService(ConfigurationService.class);
+            String maxResultWindowStr = cs.getProperty(ES_MAX_RESULT_WINDOW_PROPERTY,
+                    DEFAULT_ES_MAX_RESULT_WINDOW_VALUE);
+            try {
+                maxResultWindow = Long.valueOf(maxResultWindowStr);
+            } catch (NumberFormatException e) {
+                log.warn(String.format(
+                    "Invalid maxResultWindow property value: %s for page provider: %s, fallback to default.",
+                            maxResultWindow, getName()));
+                maxResultWindow = Long.valueOf(DEFAULT_ES_MAX_RESULT_WINDOW_VALUE);
+            }
+        }
+        return maxResultWindow;
+    }
+
+    public void setMaxResultWindow(long maxResultWindow) {
+        this.maxResultWindow = maxResultWindow;
+    }
```

Comment:
missing since version on public methods, what code is supposed to call this non-interface APIs?